File: Frontend/Academy.FE/src/app/shared/dto/spin-training.request.ts
export class SpinTrainingRequest {
  force: boolean = false;
  ecosystem: number = 0;
  account: string = "";
  trainingAssignmentSrc: string = "";
  mapping: UserTrainingMapping[] = [];
  selectedTraning: any = [];
}

export class UserTrainingMapping {
  userId: number = 0;
  userEmail: string = "";
  seniorityId: number = 0;
  seniority: string = "";
  userImage: string = "";
  trainings: ecosystemTraining[] = [];
  parent: boolean = false;
  selectedTraning: any = [];
  selected: boolean = true;
}

export class KeyValuePair<Tkey, Tval> {
  Key: Tkey | undefined;
  Value: Tval | undefined;
}

export class ecosystemTraining {
  isMvP: boolean = false;
  seniority: string = "";
  seniorityId: number = 0;
  trainingCompletionHours: number = 0;
  trainingDescription: string = "";
  trainingId: number = 0;
  trainingLink: string = "";
  trainingName: string = "";
  checked: boolean = false;
}
export class User {
  firstname: any;
  lastname: any;
  selected: any;
}

export interface Option {
  name: string;
  value: number;
  checked: boolean;
}

// export class User {
//   constructor(public firstname: string, public lastname: string, public selected?: boolean) {
//     if (selected === undefined) selected = false;
//   }
// }

File: Backend/Academy.Shared/DTO/SpinTrainingRequest.cs
namespace Academy.Shared.DTO
{
    public class SpinTrainingRequest
    {
        public bool Force { get; set; } = false;
        public int Ecosystem { get; set; } = 0;
        public string Account { get; set; }
        public string TrainingAssignmentSrc { get; set; }
        public List<UserTrainingMapping> Mapping { get; set; } = new();
        public string[] SelectedTraning { get; set; } = [];

    }

    public class UserTrainingMapping
    {
        public int UserId { get; set; } = 0;
        public string UserEmail { get; set; } = "";
        public int SeniorityId { get; set; } = 0;
        public string Seniority { get; set; } = "";
        public string UserImage { get; set; } = "";
        public List<EcosystemTraining> Trainings { get; set; } = new();
        public bool Parent { get; set; } = false;
        public string[] SelectedTraning { get; set; } = [];
    }

    public class EcosystemTraining
    {
        public bool IsMvP { get; set; } = false;
        public string Seniority { get; set; } = "";
        public int SkillId { get; set; } = 0;
        public int SeniorityId { get; set; } = 0;
        public int TrainingCompletionHours { get; set; } = 0;
        public string TrainingDescription { get; set; } = "";
        public int TrainingId { get; set; } = 0;
        public string TrainingLink { get; set; } = "";
        public string TrainingName { get; set; } = "";
    }

    public class FetchEmployeesRequest
    {
        public string Startswith { get; set; }
        public short EcosystemId { get; set; }
        public string Account { get; set; }
    }
    public class FetchDojoGexRequest
    {
        public string Startswith { get; set; }
    }
}
